akvo/rsr/static/scripts-src/project-directory.js

```diff
@@ -3,190 +3,148 @@
-      source: organisations.ttAdapter(),
-      templates: {
-        header: '<h3 class="dd-category">' + organisations_text + '</h3>',
-        suggestion: _.template('<a href="/projects/?organisation=<%= id %>"><p><%= name %></p></a>')
-       }
-    },
-    {
-      name: 'locations',
-      displayKey: 'name',
-      source: locations.ttAdapter(),
-      templates: {
-        header: '<h3 class="dd-category">' + locations_text + '</h3>',
-        suggestion: _.template('<a href="/projects/?location=<%= code %>"><p><%= name %></p></a>')
-       }
-    }
-    // ,
-    // {
-    //   name: 'sectors',
-    //   displayKey: 'name',
-    //   source: sectors.ttAdapter(),
-    //   templates: {
-    //     header: '<h3 class="dd-category">Sectors</h3>',
-    //     suggestion: _.template('<a href="/project/?sector=<%= code %>"><p><%= name %></p></a>')
-    //    }
-    // }
-  );
-
-  $('#filterForm').on('click', '.filter_focus_area', function (event) {
-    var tmp = $('#id_title').val();
-    $('select#id_focus_area option').filter(function() {
-      return $(this).text() == tmp;
-    }).prop('selected', true);
-    $('#id_title').val('');
-    $('#filterForm').submit();
-  });
-
-  $('#filterForm').on('mouseover', '.filter_focus_area', function (event) {
-    $('#collapseOne').addClass('in');
-    $('select#id_focus_area option').filter(function() {
-      return $(this).text() == $(this).val();
-    }).prop('selected', true);
-  });
-
-  $('#filterForm').on('mouseout', '.filter_focus_area', function (event) {
-    $('#collapseOne').removeClass('in');
-  });
-
-  $('#sortByDropdown').on('click', 'a', (function(event) {
-    insertParam('sort_by', $(event.target).attr('param'));
-  }));
+    var locations = new Bloodhound({
+        datumTokenizer: Bloodhound.tokenizers.obj.whitespace('name'),
+        queryTokenizer: Bloodhound.tokenizers.whitespace,
+        prefetch: {
+            //  TODO: get a filtered set of locations for Pages
+            thumbprint: "locations",
+            url: '/static/data/m49.json'
+        }
+    });
+
+    var keywords = new Bloodhound({
+        datumTokenizer: Bloodhound.tokenizers.obj.whitespace('label'),
+        queryTokenizer: Bloodhound.tokenizers.whitespace,
+        prefetch: {
+            url: '/rest/v1/typeaheads/keywords?format=json',
+            thumbprint: "keywords",
+
+            filter: function (response) {
+                return response.results;
+            }
+        }
+    });
+
+    var projects = new Bloodhound({
+        datumTokenizer: Bloodhound.tokenizers.obj.whitespace('title'),
+        queryTokenizer: Bloodhound.tokenizers.whitespace,
+        prefetch: {
+            url: '/rest/v1/typeaheads/projects?format=json&published=1',
+            thumbprint: "projects",
+
+            filter: function (response) {
+                return response.results;
+            }
+        }
+    });
+
+    projects.initialize();
+    locations.initialize();
+    keywords.initialize();
```

Comment:
Are we leaving out the organisations deliberately? Or were they missed out accidentally? 

Like I mentioned on Skype, I think, we should only have typeaheads for the project in the project directory search bar.  That will make it clear that the search is only looking in the project titles, and the typeahead is a way of making this search faster (by doing it on client side).  Similarly for the organisation and the project update directory.  The other typeahead fields are already in the advanced filter and should be used from there. 

This change, though, may need more discussion, especially with @loicsans and @AkvoAnt. Until, then I think we should keep the UI as before with the keywords typeahead added to it.